source/frontend/src/features/documents/DocumentPreviewContainer.tsx

```diff
@@ -0,0 +1,167 @@
+import moment from 'moment';
+import { FC, useCallback, useContext, useEffect, useState } from 'react';
+
+import LoadingBackdrop from '@/components/common/LoadingBackdrop';
+import { ModalContext } from '@/contexts/modalContext';
+import { ApiGen_Mayan_FilePage } from '@/models/api/generated/ApiGen_Mayan_FilePage';
+import { ApiGen_System_HttpStatusCode } from '@/models/api/generated/ApiGen_System_HttpStatusCode';
+
+import { DocumentViewerContext } from './context/DocumentViewerContext';
+import { IDocumentPreviewViewProps } from './DocumentPreviewView';
+import { createFileDownload } from './DownloadDocumentButton';
+import { useDocumentProvider } from './hooks/useDocumentProvider';
+
+export interface LoadedPages extends ApiGen_Mayan_FilePage {
+  loadedDocumentImageDataUrl?: string;
+  error: string;
+}
+
+export interface IDocumentPreviewContainerProps {
+  View: React.FC<IDocumentPreviewViewProps>;
+  createFileUrl?: (obj: Blob | MediaSource) => string;
+}
+
+export const DocumentPreviewContainer: FC<
+  React.PropsWithChildren<IDocumentPreviewContainerProps>
+> = ({ View, createFileUrl }) => {
+  const { showDocumentPreview, previewDocumentId, setShowDocumentPreview, setPreviewDocumentId } =
+    useContext(DocumentViewerContext);
+  const { setModalContent, setDisplayModal } = useContext(ModalContext);
+  const {
+    retrieveDocumentDetail,
+    downloadDocumentFilePageImage,
+    getDocumentFilePageList,
+    downloadWrappedDocumentFileLatest,
+    retrieveDocumentDetailLoading,
+    getDocumentFilePageListLoading,
+  } = useDocumentProvider();
+  const [documentPages, setDocumentPages] = useState<LoadedPages[]>([]);
+  const [currentPage, setCurrentPage] = useState<number>(0);
+
+  const loadDocumentPageImage = useCallback(
+    async (documentId: number, documentFileId: number, documentFilePageId: number) => {
+      //todo: handle image failure
+      return (createFileUrl ? createFileUrl : URL.createObjectURL)(
```

Comment:
nothing to change, but this dynamic method calling looked very odd at first glance.